libs/protocol-client/src/lib/client/contribution.ts

```diff
@@ -241,6 +241,58 @@ export class Contribution extends BaseClient {
     });
   }
 
+  public async bulkAttest(
+    networkConfig: NetworkConfig,
+    signer: ethers.Signer,
+    chainId: number,
+    userId: number,
+    data: {
+      contribution: number;
+      confidenceId: number;
+      confidenceName: string;
+      dateOfSubmission: number;
+    }[],
+  ) {
+    const args = {
+      attestations: data.map(c => {
+        return {
+          contribution: c.contribution,
+          confidence: c.confidenceId,
+          dateOfSubmission: c.dateOfSubmission,
+        };
+      }),
+    };
+
+    const contract = new GovrnContract(networkConfig, signer);
+    const transaction = await contract.bulkAttest(args);
+    const transactionReceipt = await transaction.wait();
+
+    const logs = transactionReceipt.logs.map((l, idx) => {
+      const log = contract.govrn.interface.parseLog(l);
+      return log.name === 'Attest'
+        ? {
+            index: idx,
+            contributionId: (log.args['contribution'] as BigNumber).toNumber(),
```

Comment:
What type is this? Why does it need to be explicitly set as a BigNumber?

Reply:
All `log.args`'s fields are `any` type.

Reply:
Furthermore, this is the type `args`: 
```ts
export interface Result extends ReadonlyArray<any> {
    readonly [key: string]: any;
}
```